### ISSUE TITLE
Support for Alpine.js V3

### ISSUE THREAD

**Issue by opncow** [LeanAdmin/alpine-typescript#10]
It seems `alpine-typescript` does not work with Alpine.js V3, yet. Can you please make it work with the latest Alpine.js version?

**Comment by stancl** [comment]
This package provides a bunch of wrappers that make using custom JS files easier with Alpine. In v3, this is much less needed. Honestly the only thing it could provide would be the definite type assignments for the base component class and types for the Alpine object. I think @calebporzio was thinking about adding TS definitions to Alpine 3, which would be ideal.

**Comment by opncow** [comment]
I failed using an externally defined TypeScript class with `x-data` yesterday with v3 (with and without `alpine-typescript`). The reason I really want to have the class externally is that it gets pre-processed by Webpack while my Alpine HTML does not. I think that's still a valid use-case for `alpine-typescript`, isn't it?

**Comment by stancl** [comment]
It is, but you can easily achieve it in v3 I think. Try using `(...args) => new MyClassComponent(...args)` as the value passed to `x-data`.

**Comment by opncow** [comment]
Thanks for the hint and sorry for getting back at this so late, but I was not on the computer for a few days. I'm a bit puzzled about your suggestion, sorry. For example, where do the `(...args)` in your example come from that are passed into the lambda?

I'm a bit afraid to ask, but can you please give me a short `x-data` example snippet or a reference to some code illustrating what I could try?

**Comment by stancl** [comment]
```js
class MyComponent
{
    constructor() { ... }

    // ...
}

window.foo = (...args) => new MyComponent(...args);
```

```html
<div x-data="foo('bar')">
```

Something like that maybe

**Comment by opncow** [comment]
Thanks, after some fiddling I almost got it working with this approach! 

Only issue left is that changes done to the component state by methods called on it are not picked up by Alpine. I don't understand why, though, because if I change the initial state of the component by settings something else in the constructor, I see the different initial value. So the general binding seems to work, but updates are not reflected.

**Comment by opncow** [comment]
So I could finally solve all the issues by not using TypeScript classes. Instead, I'm using interfaces and functions that return objects that implement these interfaces. With instances of TypeScript classes returned by the `window.foo` function, I ran into all sorts of issues. For example, Alpine did call this method multiple times and updates to the component state were not picked up. I tried to fix this by returning a singleton from the function, but then some `Object.defineProperty` that Alpine is doing threw errors. I could even fix that somehow, but still state changes were not picked up. I decided to just use objects in the end like the Alpine documentation shows and this worked perfectly.
